spec/features/admin/user_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+describe "a registered regular user" do
+  it "can see the same links as a visitor plus a link to login and logout" do
+    default_user = User.create(name: "Pirate Jack",
+                                  address: "123 Ocean Breeze",
+                                  city: "Bootytown",
+                                  state: "Turks & Caicos",
+                                  zip: "13375",
+                                  email: "pirate@thecarribean.com",
+                                  password: "landlubberssuck",
+                                  role: 0)
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(default_user)
```

Comment:
Curious what this does, I haven't been able to find the part in the lessons about this line even though I see it used.

Reply:
@DanielEFrampton - Copied fairly verbatim from the Authorization lesson, though it took me a bit to figure out what it was doing. I believe it's looking for 'any instance of' a user in ApplicationController and pulling :current_user from the helper method I defined there. Then it assigns it the correct admin role based on that user profile (in the test I created a user with role: 0). I believe it's a way for the test to bypass the whole login part, as you'll notice there is no place where the user is actually registered, then logging in, etc.

Reply:
Ahhh, okay. Got it. That would explain why it's not necessary in my current
user story about login. :D